web/pgadmin/settings/static/ApplicationStateProvider.jsx

```diff
@@ -0,0 +1,44 @@
+/////////////////////////////////////////////////////////////
+//
+// pgAdmin 4 - PostgreSQL Tools
+//
+// Copyright (C) 2013 - 2025, The pgAdmin Development Team
+// This software is released under the PostgreSQL Licence
+//
+//////////////////////////////////////////////////////////////
+import React, { useContext, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import getApiInstance from '../../static/js/api_instance';
+import url_for from 'sources/url_for';
+
+const ApplicationStateContext = React.createContext();
+
+export const useApplicationState = ()=>useContext(ApplicationStateContext);
+
+export function retrieveDataFromLocalStorgae(sqlId){
```

Comment:
why not move to ApplicationStateProvider with name `getToolData`?

Reply:
Done.